images/capi/packer/azure/scripts/init-sig.sh

```diff
@@ -58,46 +58,72 @@ SECURITY_TYPE_CVM_SUPPORTED_FEATURE="SecurityType=ConfidentialVmSupported"
+  plan_offer=$(jq -r '.plan_image_offer' "$target_json")
+  plan_name=$(jq -r '.plan_image_sku' "$target_json")
+  plan_version=${PLAN_VERSION:-"latest"}
+
+  # Proceed only if all plan details are valid
+  if [[ "$plan_publisher" == "null" || "$plan_offer" == "null" || "$plan_name" == "null" ]]; then
+    echo "Purchase plan details are missing. Skipping terms acceptance."
+    return
+  fi
 
-#################################################################################
-##### TODO: [SEPTEMBER 2024] Remove purchase plan info when the image is GA #####
-# Creating Azure VMs from a Marketplace Image requires a Purchase Plan
-# https://learn.microsoft.com/en-us/azure/virtual-machines/marketplace-images
-# HACK: Extract purchase plan info from the target json. We want to avoid changing the Prow jobs YAML files to add these
-# values as environment variables.
-TARGET_JSON="$(realpath packer/azure/$SIG_TARGET.json)"
-DISTRIBUTION=$(jq -r '.distribution' "$TARGET_JSON")
-DISTRIBUTION_VERSION=$(jq -r '.distribution_version' "$TARGET_JSON")
-
-plan_args=()
-if [[ "$DISTRIBUTION" == "windows" && "$DISTRIBUTION_VERSION" == "2025" ]]; then
-  PLAN_PUBLISHER=$(jq -r '.plan_image_publisher' "$TARGET_JSON")
-  PLAN_OFFER=$(jq -r '.plan_image_offer' "$TARGET_JSON")
-  PLAN_NAME=$(jq -r '.plan_image_sku' "$TARGET_JSON")
-  PLAN_VERSION=${PLAN_VERSION:-"latest"}
-
-  plan_args=(
-    --plan-name ${PLAN_NAME}
-    --plan-product ${PLAN_OFFER}
-    --plan-publisher ${PLAN_PUBLISHER}
+  # Populate the global plan_args variable
+  PLAN_ARGS=(
+    --plan-name "${plan_name}"
+    --plan-product "${plan_offer}"
+    --plan-publisher "${plan_publisher}"
   )
 
-  # Proceed only if all plan details are available
-  # WHY? Build fails with: "You have not accepted the legal terms on this subscription"
-  if [[ "$PLAN_PUBLISHER" != "null" && "$PLAN_OFFER" != "null" && "$PLAN_NAME" != "null" ]]; then
-    PLAN_URN="${PLAN_PUBLISHER}:${PLAN_OFFER}:${PLAN_NAME}:$(echo $PLAN_VERSION)"
-    echo "Plan info: ${PLAN_URN}"
+  plan_urn="${plan_publisher}:${plan_offer}:${plan_name}:${plan_version}"
 
-    # Retrieve the terms and check acceptance status
-    if [[ "$(az vm image terms show --urn "$PLAN_URN" | jq -r '.accepted')" != "true" ]]; then
-      echo "Accepting terms for image URN: ${PLAN_URN}"
-      az vm image terms accept --urn "$PLAN_URN"
-    fi
+  # Check if the image has terms to accept
+  if [[ "$(az vm image show --location "$AZURE_LOCATION" --urn "${plan_urn}" | jq -r '.plan')" == "null" ]]; then
+    echo "Image '${plan_urn}' has no terms to accept."
+    return
   fi
-fi
-############# END: SECTION TO BE REMOVED AFTER IMAGE IS GA ###################
-##############################################################################
 
+  echo "Plan info: ${plan_urn}"
+
+  # Check acceptance status and accept terms if not already accepted
+  if [[ "$(az vm image terms show --urn "$plan_urn" | jq -r '.accepted')" == "true" ]]; then
```

Comment:
```suggestion
  if [[ "$(az vm image terms show --urn "$plan_urn" -o json | jq -r '.accepted')" == "true" ]]; then
```

My local `az` client has "table" as the default output type, so this command fails unless we explicitly specify JSON output.